plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloat.hs

```diff
@@ -331,21 +348,24 @@ floatBackLets term fTable =
     floatLam lamU t = do
         herePos <- asks $ \d -> Pos d lamU LamBody
         -- make a brand new let-group comprised of all the floatable lets just inside the lam/Lam/letInTerm
-        floatAt herePos makeNewLet t
+        -- we need to force to Rec because we don't do dependency resolution at this pass, See Note [LetRec splitting pass]
```

Comment:
```suggestion
        -- we need to force to Rec because we might merge lets which depend on each other, but we can't tell because we don't do dependency resolution at this pass. So we have to be conservative. See Note [LetRec splitting pass]
```